src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java

```diff
@@ -0,0 +1,43 @@
+package edu.kis.powp.jobs2d.drivers.adapter;
+
+import edu.kis.legacy.drawer.shape.ILine;
+import edu.kis.legacy.drawer.shape.LineFactory;
+import edu.kis.legacy.drawer.panel.DrawPanelController;
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.features.DrawerFeature;
+
+
+public class LineDrawerAdapter extends DrawPanelController implements Job2dDriver{
+    private int startX = 0, startY = 0;
+
+    public LineDrawerAdapter() {
+        super();
+    }
+
+    @Override
+    public void setPosition(int x, int y) {
+        this.startX = x;
+        this.startY = y;
+    }
+
+    @Override
+    public void drawLine(ILine line)
+    {
+        DrawerFeature.getDrawerController().drawLine(line);
+    }
+
+    @Override
+    public void operateTo(int x, int y) {
+        ILine line = LineFactory.getSpecialLine();
```

Comment:
Unfortunately, this is not a generic solution. (e.g. new line type requires new class, unnecessary direct coupling with LineFactory)